src/config.rs

```diff
@@ -97,46 +200,60 @@ impl ConfigStore {
     }
 
     /// Returns the setting with the given key. If the setting is not found in the current
-    /// store, it will load the key from the storage. If the key is still not found, it will
+    /// storage, it will load the key from the storage. If the key is still not found, it will
     /// return the default value for the given key. Note that if the key is not found and no
     /// default value is specified, this function will panic.
-    pub fn get(&mut self, key: &str) -> Setting {
+    pub fn get(&self, key: &str) -> Option<Setting> {
         if !self.has(key) {
-            panic!("Setting {} not found", key);
+            return None;
         }
 
-        if let Some(setting) = self.settings.get(key) {
-            return setting.clone();
+        if let Some(setting) = self.settings.lock().unwrap().borrow().get(key) {
+            return Some(setting.clone());
         }
 
         // Setting not found, try and load it from the storage adapter
-        if let Some(setting) = self.storage.get_setting(key) {
-            self.settings.insert(key.to_string(), setting.clone());
-            return setting.clone();
+        if let Some(setting) = self.storage.get(key) {
```

Comment:
If the key is not found in the current settings, we load it from the storage, and save it in the settings for the next time.